stdlib/universe/join.go

```diff
@@ -215,6 +204,22 @@ func createMergeJoinTransformation(id execute.DatasetID, mode execute.Accumulati
 		return nil, nil, errors.New(codes.Unimplemented, "joins currently must only have two parents")
 	}
 
+	// In <https://github.com/influxdata/flux/issues/4692>, it was found that in
+	// certain cases `s.TableNames` could grow beyond the size of the parents
+	// _incorrectly_.
+	// This would lead to a panic in the for/range to follow.
+	// To illustrate the requirement in a more explicit way, this check ensures
+	// that the names of the tables, and their associated datasets are of equal counts.
+	if len(parents) != len(s.TableNames) {
```

Comment:
In case it's confusing -- the field removed in f57d93bc5bd36037c61860f3d948a358285a5c72 was the `TableNames` field of `JoinOpSpec` whereas this is the `TableNames` field on the `MergeJoinProcedureSpec`. Previously the latter was built from the former, but now it is built instead from `JoinOpSpec.params`.